jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/SessionHandler.java

```diff
@@ -257,8 +257,10 @@ public void setSecure(boolean secure)
 
         private void checkState()
         {
-            if (isStarted())
+            ServletContextHandler handler = ServletContextHandler.getCurrentServletContextHandler();
+            if (handler != null && handler.isAvailable())
```

Comment:
You need a comment here to say why `isStarted()` is not sufficient when the error message is "cannot be set after ... is started".